boa/environment.py

```diff
@@ -109,6 +109,40 @@ def patch_opcode(opcode_value, fn):
 _precompiles = {}
 
 
+def extract_arg_types(signature: str) -> str:
+    # Extract argument types and their names using regex
```

Comment:
not a huge fan of doing this using regex but i think it works here

Reply:
lol I left this comment from when I first tried to regex, but didn't get it right. I'll get GPT to write the regex